address.go

```diff
@@ -44,6 +48,39 @@ func encodeAddress(hash160 []byte, netID byte) string {
 	return base58.CheckEncode(hash160[:ripemd160.Size], netID)
 }
 
+// encodeSegWitAddress creates a bech32 encoded address string representation
+// from witness version and witness program.
+func encodeSegWitAddress(hrp string, witnessVersion byte,
+	witnessProgram []byte) (string, error) {
+
+	// Group the address bytes into 5 bit groups, as this is what is used to
+	// encode each character in the address string.
+	converted, err := bech32.ConvertBits(witnessProgram, 8, 5, true)
+	if err != nil {
+		return "", err
+	}
+
+	// Concatenate the witness version and program, and encode the resulting
+	// bytes using bech32 encoding.
+	combined := append([]byte{witnessVersion}, converted...)
+	bech, err := bech32.Encode(hrp, combined)
+	if err != nil {
+		return "", err
+	}
+
+	// Check validity by decoding the created address.
+	version, program, err := decodeSegWitAddress(hrp, bech)
```

Comment:
I see what you're getting at, I didn't really consider this method to be used for non-bitcoin addresses. If we were to make this chain agnostic, then I guess we would move Bitcoin specific code/verification into `params.chaincfg`? I'm not really sure how that would work, but would be happy to do it if I could get some guidance :)